src/fifo.rs

```diff
@@ -0,0 +1,197 @@
+//! Various functions related to FIFO
+//!
+//! TO DO:
+//! - CHECK IF ALL FUNCTIONS ARE IMPLEMENTED
+//! - MAKE SURE REGISTERS ARE NOT OVERWRITTEN BY MISTAKE
```

Comment:
Are these still TO DO?

Reply:
It should be all OK, but I will check and then remove the comments.

Reply:
Checked everything, should be OK. Removed the comments.